nil pointer dereference (`pkg/idutil/netwalker.(*NetworkWalker).Walk.func1`)

```console
$ nerdctl -v
nerdctl version 1.0.0-215-g093b10f
$ cd examples/compose-wordpress
$ nerdctl compose up -d
...
$ nerdctl compose down
INFO[0000] Removing container compose-wordpress_wordpress_1 
INFO[0000] Removing container compose-wordpress_db_1    
INFO[0000] Removing network compose-wordpress_default   
WARN[0000] error while executing [/usr/local/bin/nerdctl network rm compose-wordpress_default]: "panic: runtime error: invalid memory address or nil pointer dereference\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xfe109a]\n\ngoroutine 1 [running]:\ngithub.com/containerd/nerdctl/pkg/idutil/netwalker.(*NetworkWalker).Walk.func1(0xc000002630)\n\t/home/suda/gopath/src/github.com/containerd/nerdctl/pkg/idutil/netwalker/netwalker.go:56 +0x7a\ngithub.com/containerd/nerdctl/pkg/netutil.(*CNIEnv).FilterNetworks(0xc0000421e0?, 0xc00051fbe8)\n\t/home/suda/gopath/src/github.com/containerd/nerdctl/pkg/netutil/netutil.go:150 +0xa5\ngithub.com/containerd/nerdctl/pkg/idutil/netwalker.(*NetworkWalker).Walk(0xc00051fcc0, {0x167b818, 0xc0000ca000}, {0x7ffe90cb53a4, 0x19})\n\t/home/suda/gopath/src/github.com/containerd/nerdctl/pkg/idutil/netwalker/netwalker.go:58 +0x191\nmain.networkRmAction(0xc00030b800, {0xc0002b2a30, 0x1, 0x1?})\n\t/home/suda/gopath/src/github.com/containerd/nerdctl/cmd/nerdctl/network_rm.go:99 +0x405\ngithub.com/spf13/cobra.(*Command).execute(0xc00030b800, {0xc0000c8070, 0x1, 0x1})\n\t/home/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862\ngithub.com/spf13/cobra.(*Command).ExecuteC(0xc0004cc900)\n\t/home/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd\ngithub.com/spf13/cobra.(*Command).Execute(...)\n\t/home/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968\nmain.xmain()\n\t/home/suda/gopath/src/github.com/containerd/nerdctl/cmd/nerdctl/main.go:125 +0x9f\nmain.main()\n\t/home/suda/gopath/src/github.com/containerd/nerdctl/cmd/nerdctl/main.go:108 +0x19\n": exit status 2
```

Minimum repro
```console
$ nerdctl network create foo
2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae

$ nerdctl network rm foo
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xfe109a]

goroutine 1 [running]:
github.com/containerd/nerdctl/pkg/idutil/netwalker.(*NetworkWalker).Walk.func1(0xc00039ec60)
        /home/suda/gopath/src/github.com/containerd/nerdctl/pkg/idutil/netwalker/netwalker.go:56 +0x7a
github.com/containerd/nerdctl/pkg/netutil.(*CNIEnv).FilterNetworks(0xc0004ca0d0?, 0xc000259be8)
        /home/suda/gopath/src/github.com/containerd/nerdctl/pkg/netutil/netutil.go:150 +0xa5
github.com/containerd/nerdctl/pkg/idutil/netwalker.(*NetworkWalker).Walk(0xc000259cc0, {0x167b818, 0xc000040090}, {0x7ffe101803ba, 0x3})
        /home/suda/gopath/src/github.com/containerd/nerdctl/pkg/idutil/netwalker/netwalker.go:58 +0x191
main.networkRmAction(0xc000173200, {0xc0001c1080, 0x1, 0x1?})
        /home/suda/gopath/src/github.com/containerd/nerdctl/cmd/nerdctl/network_rm.go:99 +0x405
github.com/spf13/cobra.(*Command).execute(0xc000173200, {0xc00003e0b0, 0x1, 0x1})
        /home/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004f4900)
        /home/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
main.xmain()
        /home/suda/gopath/src/github.com/containerd/nerdctl/cmd/nerdctl/main.go:125 +0x9f
main.main()
        /home/suda/gopath/src/github.com/containerd/nerdctl/cmd/nerdctl/main.go:108 +0x19
```

(Aren't we test `nerdctl network rm` on CI? 🤔 )

--- a/pkg/idutil/netwalker/netwalker.go
+++ b/pkg/idutil/netwalker/netwalker.go
@@ -53,6 +53,10 @@ func (w *NetworkWalker) Walk(ctx context.Context, req string) (int, error) {
 	}
 
 	idFilterF := func(n *netutil.NetworkConfig) bool {
+		if n.NerdctlID == nil {
+			// External network
+			return false
+		}
 		return n.Name == req || longIDExp.Match([]byte(*n.NerdctlID)) || shortIDExp.Match([]byte(*n.NerdctlID))
 	}
 	networks, err := w.Client.FilterNetworks(idFilterF)

